proxy.go

```diff
@@ -80,8 +80,21 @@ func NewProxyFromConfig(logger *logrus.Logger, conf ProxyConfig) (p Proxy, err e
 	})
 
 	p.HTTPAddr = conf.HTTPAddress
-	// TODO Timeout?
-	p.HTTPClient = &http.Client{}
+
+	transport := &http.Transport{
+		MaxIdleConns:        conf.MaxIdleConns,
+		MaxIdleConnsPerHost: conf.MaxIdleConnsPerHost,
```

Comment:
Hmm, is it ok that these are at the zero value if unset in conf? (I think so, but would be good to confirm and add a comment maybe)!